lib/ValidateDirectiveVisitor.ts

```diff
@@ -198,9 +197,14 @@ const checkMustValidateInput = (
   }
 
   if (finalType instanceof GraphQLInputObjectType) {
-    return Object.values(finalType.getFields()).some(
-      checkMustValidateInputField,
-    );
+    const fieldsToCheck = Object.values(finalType.getFields()).filter(field => {
+      if (field.type instanceof GraphQLList) {
```

Comment:
Instead of doing this filter, can't we check this inside the checkMustValidateInput function?

Reply:
Something like that :thinking:  ?

```
const checkMustValidateInputField = (
  field: ValidatedGraphQLInputField,
): boolean => {
  // istanbul ignore next (shouldn't reach as addContainerEntryValidation would mark parent, but be safe)
  if (field.validation) {
    return true;
  }
  if (field.type instanceof GraphQLInputObjectType) {
    return !Object.values(field.type.getFields()).some(
      nestedField => nestedField.type === field.type,
    );
  }
  if (field.type instanceof GraphQLList) {
    return field.type.ofType !== field.type;
  }
  // eslint-disable-next-line @typescript-eslint/no-use-before-define
  return checkMustValidateInput(field.type);
};
```

Reply:
I just pushed something